compiler/src/org.graalvm.compiler.truffle.test/src/org/graalvm/compiler/truffle/test/inlining/InlineOnlyTest.java

```diff
@@ -0,0 +1,169 @@
+                        option("engine.CompileOnly", COMPILATION_ROOT_NAME).//
+                        option("engine.InlineOnly", METHOD_INLINED).//
+                        build()) {
+            c.eval(InlineOnlyTest.InlineSelectedMethodsTestLanguage.ID, "");
+            c.eval(InlineOnlyTest.InlineSelectedMethodsTestLanguage.ID, "");
+        }
+    }
+
+    @TruffleLanguage.Registration(id = InlineOnlyTest.ExcludePatternTestLanguage.ID, name = "InlineOnlyTest: With Exclude Pattern Test Language", version = "1.0")
+    public static class ExcludePatternTestLanguage extends ProxyLanguage {
+
+        public static final String ID = "truffle-exclude-pattern-test-language";
+
+        @Override
+        protected CallTarget parse(ParsingRequest request) throws Exception {
+            final TruffleRuntime runtime = Truffle.getRuntime();
+            final RootCallTarget rootCallTarget = runtime.createCallTarget(new RootNode(this) {
+
+                @Override
+                public String toString() {
+                    return getName();
+                }
+
+                @Override
+                public Object execute(VirtualFrame frame) {
+                    CompilerAsserts.neverPartOfCompilation("This node should not be inlined");
+                    return 99;
+                }
+
+                @Override
+                public String getName() {
+                    return METHOD_EXCLUDED_FROM_INLINING;
+                }
+            });
+            return runtime.createCallTarget(new RootNode(this) {
+
+                @Child DirectCallNode callNode = runtime.createDirectCallNode(rootCallTarget);
+
+                @Override
+                public String toString() {
+                    return getName();
+                }
+
+                @Override
+                public String getName() {
+                    return COMPILATION_ROOT_NAME;
+                }
+
+                @Override
+                public Object execute(VirtualFrame frame) {
+                    return callNode.call();
+                }
+            });
+        }
+    }
+
+    @TruffleLanguage.Registration(id = InlineOnlyTest.InlineSelectedMethodsTestLanguage.ID, name = "InlineOnlyTest: Inline Methods", version = "1.0")
+    public static class InlineSelectedMethodsTestLanguage extends ProxyLanguage {
+
+        public static final String ID = "truffle-inline-selected-methods-test-language";
+
+        @Override
+        protected CallTarget parse(ParsingRequest request) throws Exception {
+            final TruffleRuntime runtime = Truffle.getRuntime();
+            final RootCallTarget rootCallTarget = runtime.createCallTarget(new RootNode(this) {
+
+                @Override
+                public String toString() {
+                    return getName();
+                }
+
+                @Override
+                public Object execute(VirtualFrame frame) {
+                    CompilerAsserts.compilationConstant(METHOD_INLINED);
```

Comment:
This does not really assert anything.  `CompilerAsserts.compilationConstant` will not fail in the interprer so tis test will pass even if we don't inline. 
You can add a boolean flag which you can set from outside that when activated asserts that com.oracle.truffle.api.CompilerDirectives#inCompiledCode and not com.oracle.truffle.api.CompilerDirectives#inCompilationRoot